prover/prover_fri/src/gpu_prover_availability_checker.rs

```diff
@@ -36,6 +36,8 @@ pub mod availability_checker {
             stop_receiver: tokio::sync::watch::Receiver<bool>,
         ) -> anyhow::Result<()> {
             while !*stop_receiver.borrow() {
+                tokio::time::sleep(self.polling_interval).await;
```

Comment:
To keep it simple, maybe consider prover absent after receiving `None` for X times?

Reply:
I may miss some context, but it looks like an auxiliary component that simply exports some metrics and emits some logs. A proper solution that somehow really checks whether the prover is alive in real time may be much more complex than the current logic. Would it worth the effort to implement it?

Reply:
E.g. I'm not advocating for either solution, but generally I always try to find the point where idealism meets pragmatism.

Reply:
This component is created to kill the prover if it is found dead/not found in the database. We had an incident on testnet some time ago, where the prover was accidentaly marked dead which resulted in stuck proofs. I'd say it should be safe to just kill it if we received None and leave it to autoscaler to do it's work.

Reply:
In our case, on the startup sometimes it does the check before it marks itself as available in DB, which results in the crashloop(until the check is done after DB interaction, which happens after a few restarts though), so that's why this PR is created

Reply:
...which sounds like a hack on its own, so why expect it to be a "perfect hack" :) (again, I'm out of context, feel free to ignore)

Reply:
Yes, we already have discussed offline that this is more a hack which can lead to bad things and I will choose another approach to fix it